pkg/apis/build/v1alpha2/image_validation.go

```diff
@@ -89,18 +93,20 @@ func (is *ImageSpec) validateTag(ctx context.Context) *apis.FieldError {
 	return validate.Tag(is.Tag)
 }
 
-func (is *ImageSpec) validateCacheSize(ctx context.Context) *apis.FieldError {
-	if is.CacheSize != nil && ctx.Value(HasDefaultStorageClass) == nil {
-		return apis.ErrGeneric("spec.cacheSize cannot be set with no default StorageClass")
+func (is *ImageSpec) validateVolumeCache(ctx context.Context) *apis.FieldError {
+	if is.Cache.Volume != nil && ctx.Value(HasDefaultStorageClass) == nil {
```

Comment:
What is with `is.Cache != nil`?

Reply:
with the current implementation it will get defaulted, so I think it's ok